src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,336 @@ static ColliderCylinderInit D_8088BF8C = {
     },
     { 45, 77, -40, { 3310, 120, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Rock/BgHidanRock_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Rock/BgHidanRock_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -600, ICHAIN_STOP),
+};
+
+void BgHidanRock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanRock* this = THIS;
+    s32 pad;
+    CollisionHeader* colHeader = NULL;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DPM_PLAYER);
+
+    this->params = thisx->params & 0xFF;
```

Comment:
I would give this field a different name, as it's easily confused with the actor params field. `type` or something more descriptive would be better.